lib/ansible/modules/cloud/misc/terraform.py

```diff
@@ -233,7 +233,7 @@ def main():
     for k, v in variables.items():
         variables_args.extend([
             '-var',
-            '{0}={1}'.format(k, v)
+            '{0}="{1}"'.format(k, v)
```

Comment:
Hm, quoting was a major oversight when I wrote this module. Thanks for catching it! I have some thoughts on how to make this better: first, we want to quote the full value (before and after the `=`), and second I think a value might also contain quotes such as if the variable contents is a bash command for some reason.

To best handle this, we should use the `shlex` library's quote function. But to work on both python 2 and 3 we'd need `from six.moves import shlex_quote` and then this line could be:

`shlex_quote('{0}={1}'.format(k, v))`